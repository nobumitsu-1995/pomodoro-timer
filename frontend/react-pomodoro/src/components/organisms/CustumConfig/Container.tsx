import React, { useEffect, useState } from 'react'
import { useDispatch } from 'react-redux'
import { tokenGetSelector } from 'src/feature/selectors/token'
import { custumConfigsSelector } from 'src/feature/selectors/custumConfig'
import { updateCustumConfig } from 'src/feature/slices/custumConfig'
import { setCustumTimerConfig } from 'src/feature/slices/timerConfig'
import { useSelector } from 'src/feature/store'
import { api } from 'src/lib/functions/axios'
import {
  validateCycle,
  validateCycleToLongRestTime,
  validateTimerConfig,
} from 'src/lib/functions/validation'
import Presenter from './Presenter'

const Container: React.FC = () => {
  const dispatch = useDispatch()
  const token = useSelector(tokenGetSelector)
  const custumConfigs = useSelector(custumConfigsSelector)
  const [selectNum, setSelectNum] = useState(0)
  const [custumConfig, setCustumConfig] = useState(custumConfigs[0])
  const [errors, setErrors] = useState({
    workTime: '',
    restTime: '',
    cycle: '',
    longRestTime: '',
    cycleToLongRestTime: '',
  })

  useEffect(() => {
    setCustumConfig(custumConfigs[selectNum])
    dispatch(setCustumTimerConfig(custumConfigs[selectNum]))
  }, [selectNum, custumConfigs])

  const formItems = [
    {
      id: 'workTime',
      label: 'work time',
      value: custumConfig.workTime,
      error: errors.workTime || '',
    },
    {
      id: 'restTime',
      label: 'rest time',
      value: custumConfig.restTime,
      error: errors.restTime || '',
    },
    {
      id: 'cycle',
      label: 'cycle',
      value: custumConfig.cycle,
      error: errors.cycle || '',
    },
    {
      id: 'longRestTime',
      label: 'long rest time',
      value: custumConfig.longRestTime,
      error: errors.longRestTime || '',
    },
    {
      id: 'cycleToLongRestTime',
      label: (
        <>
          Cycles to <br /> long rest periods
        </>
      ),
      value: custumConfig.cycleToLongRestTime,
      error: errors.cycleToLongRestTime || '',
    },
  ]

  const changeEdit = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = event.target
    validateForm(name, Number(value))
    setCustumConfig((prev) => ({
      ...prev,
      [name]: value,
    }))
  }

  const validateForm = (name: string, value: number) => {
    switch (name) {
      case 'cycle':
        setErrors((prev) => ({
          ...prev,
          [name]: validateCycle(value),
        }))
        break
      case 'cycleToLongRestTime':
        setErrors((prev) => ({
          ...prev,
          [name]: validateCycleToLongRestTime(value),
        }))
        break
      case 'workTime':
      case 'restTime':
      case 'longRestTime':
        setErrors((prev) => ({
          ...prev,
          [name]: validateTimerConfig(value),
        }))
        break
      default:
        break
    }
  }

  const clickUpdate = () => {
    if (checkFormsValidaty()) return

    api(token)
      .patch(`/v1/custum_config/${custumConfig._id}/update`, custumConfig)
      .then((res) => {
        dispatch(updateCustumConfig(res.data))
        dispatch(setCustumTimerConfig(res.data))
      })
      .catch((e) => {
        console.error(e)
      })
  }

  const checkFormsValidaty = () => {
    Object.entries(custumConfig).forEach((data) => {
      validateForm(data[0], Number(data[1]))
    })

    return (
      Object.values(errors).filter((value) => {
        return value !== ''
      }).length !== 0
    )
  }

  const onChangeSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
    const { value } = event.target
    setSelectNum(parseInt(value))
  }

  return (
    <Presenter
      formItems={formItems}
      configLength={5}
      clickUpdate={clickUpdate}
      changeEdit={changeEdit}
      onChangeSelect={onChangeSelect}
    />
  )
}

export default Container
