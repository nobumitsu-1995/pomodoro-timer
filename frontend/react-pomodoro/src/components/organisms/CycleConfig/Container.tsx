import React, { useCallback, useEffect, useState } from 'react'
import { useDispatch } from 'react-redux'
import { cycleSelector } from '../../../feature/selectors/timerConfig'
import { useSelector } from '../../../feature/store'
import { updateCycle as reduxUpdateCycle } from '../../../feature/slices/timerConfig'
import Presenter from './Presenter'
import { validateCycle } from 'src/lib/functions/validation'

const Container: React.FC = () => {
  const dispatch = useDispatch()
  const updateCycle = useCallback(
    (cycle: number) => {
      dispatch(reduxUpdateCycle(cycle))
    },
    [dispatch]
  )
  const globalCycle = useSelector(cycleSelector)
  const [cycle, setCycle] = useState(globalCycle)
  const [error, setError] = useState('')

  useEffect(() => {
    setCycle(globalCycle)
  }, [globalCycle])

  const handleChangeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
    const _cycle = Number(e.target.value)
    setError(validateCycle(_cycle))
    setCycle(Number(_cycle))
  }

  const handleClickButton = () => {
    const _error = validateCycle(cycle)
    setError(_error)
    if (_error) return
    updateCycle(cycle)
  }

  const handleOnClickUp = () => {
    const _cycle = cycle + 1
    const _error = validateCycle(_cycle)
    setError(_error)

    if (_error) return
    setCycle(_cycle)
    updateCycle(_cycle)
  }

  const handleOnClickDown = () => {
    const _cycle = cycle - 1
    const _error = validateCycle(_cycle)
    setError(_error)

    if (_error) return
    setCycle(_cycle)
    updateCycle(_cycle)
  }

  return (
    <Presenter
      value={cycle}
      error={error}
      onChange={handleChangeInput}
      onClick={handleClickButton}
      onClickUp={handleOnClickUp}
      onClickDown={handleOnClickDown}
    />
  )
}

export default Container
