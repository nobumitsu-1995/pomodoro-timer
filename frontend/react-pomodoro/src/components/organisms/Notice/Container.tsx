import React, { useState } from 'react'
import { noticesSelector } from '../../../feature/selectors/notice'
import { useSelector } from '../../../feature/store'
import { NoticeType } from '../../../lib/types/modelType'
import Presenter from './Presenter'

const Container: React.FC = () => {
  const notices = useSelector(noticesSelector)
  const [currentNotice, setCurrentNotice] = useState<NoticeType | undefined>()

  return (
    <Presenter
      notices={notices}
      notice={currentNotice}
      setCurrentNotice={setCurrentNotice}
    />
  )
}

export default Container
