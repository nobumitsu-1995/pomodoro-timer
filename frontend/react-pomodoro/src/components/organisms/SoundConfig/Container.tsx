import React from 'react'
import useSoundConfig from 'src/lib/hooks/useSoundConfig'
import { volumeSelector } from '../../../feature/selectors/soundConfig'
import { useSelector } from '../../../feature/store'
import Presenter from './Presenter'

const Container: React.FC = () => {
  const volume = useSelector(volumeSelector)
  const globalVolume = useSelector(volumeSelector)
  const { playRestFinish, updateVolume, unMuteVolume, muteVolume } =
    useSoundConfig(volume)

  const handleChangeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
    const _volume = Number(e.target.value)
    updateVolume(_volume)
  }

  const handleClickButton = () => {
    if (globalVolume === 0) {
      unMuteVolume()
    } else {
      muteVolume()
    }
  }

  return (
    <Presenter
      isMuted={globalVolume === 0}
      onClick={handleClickButton}
      onChange={handleChangeInput}
      value={globalVolume}
      onClickTry={playRestFinish}
    />
  )
}

export default Container
