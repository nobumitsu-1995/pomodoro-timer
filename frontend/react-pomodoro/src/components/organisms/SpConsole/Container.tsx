import { useAuth0 } from '@auth0/auth0-react'
import React, { useState } from 'react'
import Console from 'src/components/templates/Console'
import { Notice } from '..'
import UserAchievement from '../UserAchievement/Container'
import Presenter from './Presenter'

const Container = () => {
  const [isOpen, setIsOpen] = useState(false)
  const [content, setContent] = useState<string>('')
  const { isAuthenticated } = useAuth0()

  const touchButton = (id: string) => {
    if (id === content) {
      return setIsOpen(!isOpen)
    }
    if (isOpen) {
      return setContent(id)
    }
    setContent(id)
    setIsOpen(!isOpen)
  }

  const CONTENT_LIST = [
    {
      id: 'notice',
      content: <Notice />,
    },
    {
      id: 'help',
      content: <Notice />,
    },
    {
      id: 'config',
      content: <Console />,
    },
    {
      id: 'user',
      content: <UserAchievement />,
    },
  ]

  return (
    <Presenter
      isLoggedIn={isAuthenticated}
      isOpen={isOpen}
      content={CONTENT_LIST.find((v) => v.id === content)?.content}
      touchButton={touchButton}
    />
  )
}

export default Container
