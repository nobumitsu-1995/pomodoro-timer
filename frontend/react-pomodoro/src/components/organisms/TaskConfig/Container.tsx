import React, { useState } from 'react'
import { TaskType } from 'src/lib/types/modelType'
import { useDispatch } from 'react-redux'
import { useSelector } from 'src/feature/store'
import { currentTaskSelector, tasksSelector } from 'src/feature/selectors/task'
import {
  addTask,
  setCurrentTask,
  deleteTask,
  updateTask,
} from 'src/feature/slices/tasks'
import { api } from '../../../lib/functions/axios'
import Presenter from './Presenter'
import { tokenGetSelector } from 'src/feature/selectors/token'

const Container: React.FC = () => {
  const dispatch = useDispatch()
  const currentTask = useSelector(currentTaskSelector)
  const token = useSelector(tokenGetSelector)
  const tasks = useSelector(tasksSelector)

  const [task, setTask] = useState({
    title: '',
    id: '',
  })
  const [isEditMode, setIsEditMode] = useState(false)

  const onClickTask = (task: TaskType) => {
    dispatch(setCurrentTask(task))
  }

  const onClickDeleteTask = (id: string) => {
    if (!confirm('削除してよろしいですか？')) return
    api(token)
      .delete(`/v1/task/${id}/delete`)
      .then(() => {
        dispatch(deleteTask(id))
        setTask({ title: '', id: '' })
      })
      .catch((e) => {
        console.error(e)
      })
  }

  const onClickEditTask = (task: TaskType) => {
    setIsEditMode(true)
    setTask({ title: task.title, id: task._id })
  }

  const onClickUpdateTask = () => {
    if (task.title.length === 0) return

    api(token)
      .patch(`/v1/task/${task.id}/update`, {
        title: task.title,
      })
      .then((res) => {
        dispatch(updateTask(res.data))
        dispatch(setCurrentTask(res.data))
        setTask({ title: '', id: '' })
        setIsEditMode(false)
      })
      .catch((e) => {
        console.error(e)
      })
  }

  const onClickCreateTask = () => {
    if (task.title.length === 0) return

    api(token)
      .post('/v1/task/create', {
        title: task.title,
      })
      .then((res) => {
        dispatch(addTask(res.data))
        dispatch(setCurrentTask(res.data))
        setTask({ title: '', id: '' })
      })
      .catch((e) => {
        console.error(e)
      })
  }

  const onChangeInput = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value } = event.target
    setTask({
      ...task,
      title: value,
    })
  }

  return (
    <Presenter
      task={task}
      tasks={tasks}
      currentTask={currentTask}
      isEditMode={isEditMode}
      onClickTask={onClickTask}
      onClickDeleteTask={onClickDeleteTask}
      onClickEditTask={onClickEditTask}
      onClickUpdateTask={onClickUpdateTask}
      onClickCreateTask={onClickCreateTask}
      onChangeInput={onChangeInput}
    />
  )
}

export default Container
