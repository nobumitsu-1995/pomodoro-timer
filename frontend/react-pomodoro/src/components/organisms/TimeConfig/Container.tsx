import React, { useCallback, useState } from 'react'
import {
  incrementWorkTime as reduxIncrementWorktime,
  decrementWorkTime as reduxDecrementWorktime,
  incrementRestTime as reduxIncrementResttime,
  decrementRestTime as reduxDecrementResttime,
} from '../../../feature/slices/timerConfig'
import { useDispatch } from 'react-redux'
import { useSelector } from '../../../feature/store'
import {
  restTimeSelector,
  workTimeSelector,
} from '../../../feature/selectors/timerConfig'
import Presenter from './Presenter'

const Container: React.FC = () => {
  const [error, setError] = useState({ work: '', rest: '' })
  const dispatch = useDispatch()

  const globalWorkTime = useSelector(workTimeSelector) / 60
  const globalRestTime = useSelector(restTimeSelector) / 60

  const incrementWorkTime = useCallback(() => {
    if (globalWorkTime === 60) {
      return setError({ ...error, work: 'too big' })
    }
    setError({ ...error, work: '' })
    dispatch(reduxIncrementWorktime())
  }, [dispatch, globalWorkTime])

  const decrementWorkTime = useCallback(() => {
    if (globalWorkTime === 5) {
      return setError({ ...error, work: 'too small' })
    }
    setError({ ...error, work: '' })
    dispatch(reduxDecrementWorktime())
  }, [dispatch, globalWorkTime])

  const incrementRestTime = useCallback(() => {
    if (globalRestTime === 60) {
      return setError({ ...error, rest: 'too big' })
    }
    setError({ ...error, rest: '' })
    dispatch(reduxIncrementResttime())
  }, [dispatch, globalRestTime])

  const decrementRestTime = useCallback(() => {
    if (globalRestTime === 5) {
      return setError({ ...error, rest: 'too small' })
    }
    setError({ ...error, rest: '' })
    dispatch(reduxDecrementResttime())
  }, [dispatch, globalRestTime])

  const consoleItems = [
    {
      icon: 'work',
      title: 'work\ntime',
      value: globalWorkTime,
      error: error.work,
      onClickUp: incrementWorkTime,
      onClickDown: decrementWorkTime,
    },
    {
      icon: 'local_cafe',
      title: 'rest\ntime',
      value: globalRestTime,
      error: error.rest,
      onClickUp: incrementRestTime,
      onClickDown: decrementRestTime,
    },
  ]

  return <Presenter consoleItems={consoleItems} />
}

export default Container
