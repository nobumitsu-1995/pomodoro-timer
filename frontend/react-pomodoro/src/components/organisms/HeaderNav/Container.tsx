import React from 'react'
import { useAuth0 } from '@auth0/auth0-react'
import Presenter from './Presenter'
import Notice from '../Notice/Container'
import UserAchievement from '../UserAchievement/Container'

const Container: React.FC = () => {
  const { isAuthenticated } = useAuth0()

  const listButtons = [
    {
      modalContent: <UserAchievement />,
      icon: 'person',
      isHide: !isAuthenticated,
    },
    {
      modalContent: <Notice />,
      icon: 'notifications',
    },
    {
      modalContent: <p>Help Page</p>,
      icon: 'help',
    },
  ]

  return <Presenter listButtons={listButtons} isLoggedIn={isAuthenticated} />
}

export default Container
