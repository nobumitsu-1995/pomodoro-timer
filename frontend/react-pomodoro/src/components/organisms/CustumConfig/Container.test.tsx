/* eslint-disable react/react-in-jsx-scope */
import { fireEvent, render } from '@testing-library/react'
import Container from './Container'
import { Provider } from 'react-redux'
import { store } from 'src/feature/store'
import Theme from 'src/assets/styles/Theme'
import {
  validateCycle,
  validateCycleToLongRestTime,
  validateTimerConfig,
} from 'src/lib/functions/validation'
import axios from 'axios'
import { initialState } from 'src/feature/slices/custumConfig'
import { userEvent } from '@storybook/testing-library'

jest.mock('axios')
jest.mock('src/lib/functions/validation', () => ({
  validateCycle: jest.fn(),
  validateTimerConfig: jest.fn(),
  validateCycleToLongRestTime: jest.fn(),
}))

const mockedAxios = axios as jest.Mocked<typeof axios>
const mockedValidateCycle = jest.mocked(validateCycle)
const mockedValidateCycleToLongRestTime = jest.mocked(
  validateCycleToLongRestTime
)
const mockedValidateTimerConfig = jest.mocked(validateTimerConfig)

describe('CustumConfig', () => {
  afterEach(() => {
    mockedValidateCycle.mockReset()
    mockedValidateTimerConfig.mockReset()
    mockedValidateCycleToLongRestTime.mockReset()
    mockedAxios.patch.mockReset()
  })

  test('snapshot', () => {
    const tree = render(
      <Provider store={store}>
        <Theme>
          <Container />
        </Theme>
      </Provider>
    ).asFragment()
    expect(tree).toMatchSnapshot()
  })

  test('コンポーネント機能の使用テスト', () => {
    mockedAxios.create.mockReturnValueOnce(mockedAxios)
    mockedAxios.patch.mockResolvedValueOnce({
      data: initialState.custumConfig[0],
    })
    mockedValidateCycle.mockReturnValue('')
    mockedValidateTimerConfig.mockReturnValue('')
    mockedValidateCycleToLongRestTime.mockReturnValue('')

    const { getByText, getByLabelText, getByRole } = render(
      <Provider store={store}>
        <Theme>
          <Container />
        </Theme>
      </Provider>
    )

    const workTime = getByLabelText('work time')
    const restTime = getByLabelText('rest time')
    const cycle = getByLabelText('cycle')
    const longRestTime = getByLabelText('long rest time')
    const cycleToLongRestTime = getByLabelText('Cycles to long rest periods')

    expect(workTime).toHaveValue(5)
    expect(restTime).toHaveValue(5)
    expect(cycle).toHaveValue(1)
    expect(longRestTime).toHaveValue(5)
    expect(cycleToLongRestTime).toHaveValue(1)

    // formの入力
    userEvent.type(workTime, '{backspace}25')
    userEvent.type(restTime, '{backspace}10')
    userEvent.type(cycle, '{backspace}5')
    userEvent.type(longRestTime, '{backspace}15')
    userEvent.type(cycleToLongRestTime, '{backspace}3')

    expect(workTime).toHaveValue(25)
    expect(restTime).toHaveValue(10)
    expect(cycle).toHaveValue(5)
    expect(longRestTime).toHaveValue(15)
    expect(cycleToLongRestTime).toHaveValue(3)

    // 入力情報の送信
    const btn = getByText('UPDATE')
    fireEvent.click(btn)
    expect(mockedValidateCycle).toBeCalledTimes(3)
    expect(mockedValidateCycleToLongRestTime).toBeCalledTimes(3)
    expect(mockedValidateTimerConfig).toBeCalledTimes(12)
    expect(mockedAxios.patch).toBeCalledWith('/v1/custum_config/0/update', {
      _id: '0',
      cycle: '5',
      cycleToLongRestTime: '3',
      longRestTime: '15',
      restTime: '10',
      workTime: '25',
    })

    // セレクターの変更
    const selector = getByRole('combobox')
    expect(selector).toHaveValue('0')
    userEvent.selectOptions(selector, '4')
    expect(selector).toHaveValue('4')
    expect(workTime).toHaveValue(5)
    expect(restTime).toHaveValue(5)
    expect(cycle).toHaveValue(1)
    expect(longRestTime).toHaveValue(5)
    expect(cycleToLongRestTime).toHaveValue(1)
  })
})
