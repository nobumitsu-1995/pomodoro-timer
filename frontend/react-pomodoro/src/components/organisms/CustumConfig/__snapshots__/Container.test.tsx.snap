// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustumConfig snapshot 1`] = `
<DocumentFragment>
  <div
    class="Paper__StyledDiv-sc-eiawt3-0 Paper___StyledStyledDiv-sc-eiawt3-1 mEzeE jtvgsP"
  >
    <div
      class="sc-ftvSup NVXIp"
    >
      <p
        class="Text__StyledP-sc-5vvbrk-0 Text___StyledStyledP-sc-5vvbrk-1  hdfbys"
      >
        Custum Timer Edit
      </p>
      <div
        class="sc-bczRLJ ehiAyy"
      >
        <span
          class="sc-gsnTZi clrkUc"
        />
        <select
          class="sc-dkzDqf bgaUQm"
        >
          <option
            value="0"
          >
            option1
          </option>
          <option
            value="1"
          >
            option2
          </option>
          <option
            value="2"
          >
            option3
          </option>
          <option
            value="3"
          >
            option4
          </option>
          <option
            value="4"
          >
            option5
          </option>
        </select>
      </div>
    </div>
    <form
      class="sc-papXJ bdgcHk"
    >
      <div
        class="sc-hKMtZM eOlEBc sc-gKXOVf hIWLdE"
      >
        <label
          for="workTime"
        >
          <p
            class="Text__StyledP-sc-5vvbrk-0 Text___StyledStyledP-sc-5vvbrk-1  dUAuAw"
          >
            work time
          </p>
        </label>
        <div
          class="sc-eCYdqJ kVYsYd"
        >
          <input
            class="Input__StyledInput-sc-1uzn9jx-0 Input___StyledStyledInput-sc-1uzn9jx-1 iioXxK ihOHkk"
            id="workTime"
            name="workTime"
            placeholder="Number"
            type="number"
            value="5"
          />
        </div>
      </div>
      <div
        class="sc-hKMtZM eOlEBc sc-gKXOVf hIWLdE"
      >
        <label
          for="restTime"
        >
          <p
            class="Text__StyledP-sc-5vvbrk-0 Text___StyledStyledP-sc-5vvbrk-1  dUAuAw"
          >
            rest time
          </p>
        </label>
        <div
          class="sc-eCYdqJ kVYsYd"
        >
          <input
            class="Input__StyledInput-sc-1uzn9jx-0 Input___StyledStyledInput-sc-1uzn9jx-1 iioXxK ihOHkk"
            id="restTime"
            name="restTime"
            placeholder="Number"
            type="number"
            value="5"
          />
        </div>
      </div>
      <div
        class="sc-hKMtZM eOlEBc sc-gKXOVf hIWLdE"
      >
        <label
          for="cycle"
        >
          <p
            class="Text__StyledP-sc-5vvbrk-0 Text___StyledStyledP-sc-5vvbrk-1  dUAuAw"
          >
            cycle
          </p>
        </label>
        <div
          class="sc-eCYdqJ kVYsYd"
        >
          <input
            class="Input__StyledInput-sc-1uzn9jx-0 Input___StyledStyledInput-sc-1uzn9jx-1 iioXxK ihOHkk"
            id="cycle"
            name="cycle"
            placeholder="Number"
            type="number"
            value="1"
          />
        </div>
      </div>
      <div
        class="sc-hKMtZM eOlEBc sc-gKXOVf hIWLdE"
      >
        <label
          for="longRestTime"
        >
          <p
            class="Text__StyledP-sc-5vvbrk-0 Text___StyledStyledP-sc-5vvbrk-1  dUAuAw"
          >
            long rest time
          </p>
        </label>
        <div
          class="sc-eCYdqJ kVYsYd"
        >
          <input
            class="Input__StyledInput-sc-1uzn9jx-0 Input___StyledStyledInput-sc-1uzn9jx-1 iioXxK ihOHkk"
            id="longRestTime"
            name="longRestTime"
            placeholder="Number"
            type="number"
            value="5"
          />
        </div>
      </div>
      <div
        class="sc-hKMtZM eOlEBc sc-gKXOVf hIWLdE"
      >
        <label
          for="cycleToLongRestTime"
        >
          <p
            class="Text__StyledP-sc-5vvbrk-0 Text___StyledStyledP-sc-5vvbrk-1  dUAuAw"
          >
            Cycles to 
            <br />
             long rest periods
          </p>
        </label>
        <div
          class="sc-eCYdqJ kVYsYd"
        >
          <input
            class="Input__StyledInput-sc-1uzn9jx-0 Input___StyledStyledInput-sc-1uzn9jx-1 iioXxK ihOHkk"
            id="cycleToLongRestTime"
            name="cycleToLongRestTime"
            placeholder="Number"
            type="number"
            value="1"
          />
        </div>
      </div>
      <button
        class="Button__StyledButton-sc-mes1hd-0 Button___StyledStyledButton-sc-mes1hd-1 fxBFHR cFTvDQ sc-iBkjds bbiPIP"
        type="button"
      >
        UPDATE
      </button>
    </form>
  </div>
</DocumentFragment>
`;
